### PR TITLE
Don't secretly install mysql gem

### LINKED ISSUE #76: mysql gem built even when not necessary
If I include `marginalia` in a Gemfile, even if I don't use/need `mysql`, it looks like the gem will always be built on ruby < 2.4.
It seems like there's no differentiation between installing `marginalia` as a dependency and for development in case of that extension.

This is causing an issue, because I can't successfully build the `mysql` gem on the current system.

### PR DESCRIPTION
**Pull request by orien**
Version 1.6.0 is secretly installing the `mysql` gem in production environments running Ruby 2.3 and lower via the `mkrf_conf` extension. This is a surprising behaviour.

From the looks of #69 the intent was for `mysql` to be installed when developing `marginalia` rather a requirement for all downstream systems.

### Change

Remove the `mkrf_conf` extension and define the `mysql` gem dependency in the Gemfile. This way the dependency will not leak to downstream systems.

Also Fixes #76.

### CODE REVIEW

File: Rakefile [hunk 1]
```diff
@@ -5,7 +5,12 @@ task :default => ['test:all']
 
 namespace :test do
   desc "test all drivers"
-  task :all => [:mysql2, :postgresql, :sqlite]
+  begin
+    require 'mysql'
+    task :all => [:mysql, :mysql2, :postgresql, :sqlite]
+  rescue LoadError
+    task :all => [:mysql2, :postgresql, :sqlite]
+  end
```
Comment (arthurnn) [inline Rakefile]: I dropped mysql support on tests. https://github.com/basecamp/marginalia/commit/e03788f4e8a90df5478fa920e4c765012be13eb5

I think we can remove the require. do you need it?

Comment (orien) [reply Rakefile]: I don't need it. I'll remove this change.

**Comment by arthurnn** [comment]
Thanks

**Comment by alyssais** [comment]
Any change of a release with this change? I've seen a couple of developers confused by the marginalia install failing, especially given the lack of error message, since it just swallowed the exception.

**Pull request merged**